compiler/noirc_printable_type/src/lib.rs

```diff
@@ -253,8 +253,15 @@ fn to_string<F: AcirField>(value: &PrintableValue<F>, typ: &PrintableType) -> Op
     Some(output)
 }
 
-// Taken from Regex docs directly
-fn replace_all<E>(
+/// Assumes `re` is a regex that matches interpolation sequences like `{...}`.
+/// Replaces all matches by invoking `replacement`, but only if a match isn't preceded by an odd
+/// number of '{'.
+/// For example, if we have `{{x}}` then there's one `{` before the interpolation, and we can
+/// see it's actually not an interpolation.
+/// However, if we have `{{{x}}}` there are two `{` before the interpolation, so the third
+/// `{` actually starts an interpolation.
+/// With four (`{{{{x}}}}`) it's again even and it's not an interpolation, etc.
+fn replace_all_interpolations<E>(
```

Comment:
So previously this was a general `replace_all` function that could replace all captures in any regex, but now it's specific to format strings because it knows that it has to count the number of curly braces. 

I would suggest to go either of two ways:
* Remove the `re` parameter and create the regex looking for interpolations inside the function. I don't see a reason why it should _assume_ that it's given a regex that does the right thing. It's called from one place anyway, so there is no need for that regex to be created there, when it's the only one a `replace_all_interpolations` can probably work with anyway. 
* Leave it as `replace_all` and move the curly brace counting logic into the `replacement` function implementation. the `Captures` has access to the `haystack`, so I think it can do the same there as well, except you have to return an actual string to "leave" an odd number of stuff in place.

Or do both, to leave the generic `replace_all` around and also have an interpolation specific version.

Reply:
I made it work without regex. The code is likely a bit simpler and also faster (maybe?), and I added a few tests too.

Reply:
You da man!